packages/react-api/src/hooks/useViewportFeatures.js

```diff
@@ -26,6 +26,13 @@ export default function useViewportFeatures(
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const computeFeaturesTileset = useCallback(
     debounce(async ({ tiles, viewport, uniqueIdProperty, sourceId }) => {
+      dispatch(
+        setViewportFeaturesReady({
+          sourceId,
+          ready: false
+        })
+      );
```

Comment:
Was the lack of this dispatch here causing a bug in calculations?

Reply:
Not a bug, but misinformation. I didn't know when the source wasn't ready. Other widgets know when to change using widget loading state, but that cannot be applied in time series because it's applying a filter constantly and then the widget was updating all the time. That's why I need to know when the source is ready or not.

Reply:
I don't know if I understand you very well. If the time series it's applying a filter constantly it's like changing any other filter so you don't need setViewportFeaturesReady to false, right? Also, I think this would change the flow in the rest of the widgets

Reply:
Yes, in any other widget, when it applies a filter, the useEffect for fetching data is executed to get the exact same data. It doesn't happen in TimeSeries widget, it only fetches data when: A filter change or source's ready state changes from false to true.